### ISSUE TITLE
Training parameter on different datasets

### ISSUE THREAD

**Issue by kinfeparty** [neuroailab/VIE#13]
Hello! I meet a new question.
the paper's training dataset is kinetic dataset which contains 400 labels and about 240000 videos. I found that the parameter of kmeans_k is 10000 in your training strategy. 
I want to train the model in jester dataset which contains 27 labels and about 120000 videos. Which is smaller and simpler than kinetic dataset.
I train VIE-3DResNet-IR training as pretraining for the VIE-3DResNet for 10 epoch following the tutorial.The top NN result is 10% .However the VIE-3DResNet's result is not good .No matter I change the kmeans_k value. The result can't reach 10% and the accuracy decrease .
However if I continue training the IR model, The top NN result can reach 60%.

Different datasets have different training parameters.I use the baseline loss function but get the better result which confuse me. Can you provide some advice for me or is there more parameter I forget to change?

**Comment by chengxuz** [comment]
Yeah, you are right that this kmeans_k parameter indeed is dependent on the dataset and requires tuning on new datasets. I usually scale it according to the number of items in the dataset, so in your case, it would be 5000. However, it is also dependent on how complicated this dataset is. If the videos in the dataset are very similar to each other, which reduces the complexity, you need to further reduce this number, like to 3000 or even lower. To achieve even better results, you can use combinations of multiple clusters, which can be done through setting this parameter as `10000,10000,10000,10000,10000,10000`, the number of clusters is equal to how many numbers in your kmeans_k, I would also try this one.

Although I am confident that after properly tuning this parameter, VIE-LA can reach better results than VIE-IR, jus as reported in the paper, you should feel free to just use VIE-IR model if you don't have enough time/resources to do this parameter search.

**Comment by kinfeparty** [comment]
Thanks for your great answering!
